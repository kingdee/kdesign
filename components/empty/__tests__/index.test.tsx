import React from 'react'
import { render, mount } from 'enzyme'
import Empty from '../index'
import mountTest from '../../../tests/shared/mountTest'

describe('Empty', () => {
  // 1. mount test
  mountTest(() => <Empty />)

  // 2. render test
  it('renders correctly', () => {
    expect(render(<Empty />)).toMatchSnapshot()

    expect(render(<Empty description="我是空组件" />)).toMatchSnapshot()

    expect(render(<Empty image="http://ikd.kingdee.com/ikd2/images/common/empty-content.png" />)).toMatchSnapshot()

    expect(render(<Empty>空组件自定义附属内容</Empty>)).toMatchSnapshot()
  })

  // 3. warns in component

  // 4. render null or undefined without errors
  it('render null or undefined spin without errors', () => {
    expect(
      mount(
        <Empty image={false} description={false}>
          {null}
          {undefined}
        </Empty>,
      ),
    ).not.toBeEmptyRender()
  })

  // 5. displayName
  it('should have displayName static property', () => {
    const wrapper = mount(<Empty />)
    expect((wrapper.type() as any).displayName).toBe('Empty')
  })

  // 6. class state
  it('should class use right', () => {
    const defaultEmpty = mount(<Empty />)
    expect(defaultEmpty.find(`.kd-empty`)).toExist()
    expect(defaultEmpty.find(`.kd-empty-image`)).toExist()
    expect(defaultEmpty.find(`.kd-empty-description`)).toExist()

    const noImageEmpty = mount(<Empty image={false} />)
    expect(noImageEmpty.find(`.kd-empty-image`)).not.toExist()
    expect(noImageEmpty.find(`.kd-empty-description`)).toExist()

    const noDescriptionEmpty = mount(<Empty description={false} />)
    expect(noDescriptionEmpty.find(`.kd-empty-image`)).toExist()
    expect(noDescriptionEmpty.find(`.kd-empty-description`)).not.toExist()

    const hasChildrenEmpty = mount(<Empty>点击新增</Empty>)
    expect(hasChildrenEmpty.find(`.kd-empty-footer`)).toExist()
    expect(hasChildrenEmpty.find(`.kd-empty-footer`).text()).toEqual('点击新增')

    const customDescriptionEmpty = mount(<Empty description="自定义描述" />)
    expect(customDescriptionEmpty.find(`.kd-empty-description`).text()).toEqual('自定义描述')

    const customImageEmpty = mount(<Empty image="http://ikd.kingdee.com/ikd2/images/common/empty-content.png" />)
    expect(customImageEmpty.find(`.kd-empty-image`).prop('src')).toEqual(
      'http://ikd.kingdee.com/ikd2/images/common/empty-content.png',
    )
  })

  // 7. event

  // 8. config provider

  // 9. ref test
  it('should get Empty element from ref', () => {
    const ref = React.createRef()
    mount(<Empty ref={ref} />)
    expect(ref.current instanceof HTMLDivElement).toBe(true)
  })

  // 10. api test
  it('api test', () => {
    const wrapper = mount(<Empty imageStyle={{ height: '200px' }} />)
    expect(wrapper.find('.kd-empty-image').props().style?.height).toEqual('200px')

    const wrapperSrc = mount(<Empty image="" imageStyle={{ height: '200px' }} />)
    expect(wrapperSrc.find('.kd-empty-image').props().style?.height).toEqual('200px')
  })
})
