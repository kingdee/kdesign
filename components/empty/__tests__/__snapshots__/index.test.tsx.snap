// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty renders correctly 1`] = `
<div
  class="kd-empty"
>
  <svg
    class="kd-empty-image"
    height="96px"
    version="1.1"
    viewBox="0 0 168 96"
    width="168px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      编组
    </title>
    <defs>
      <linearGradient
        id="kd-empty-default-linearGradient3"
        x1="50%"
        x2="50%"
        y1="0%"
        y2="100%"
      >
        <stop
          offset="0%"
          stop-color="#D6DCE7"
        />
        <stop
          offset="100%"
          stop-color="#D6DCE7"
          stop-opacity="0"
        />
      </linearGradient>
    </defs>
    <g
      fill="none"
      fill-rule="evenodd"
      id="页面-1"
      stroke="none"
      stroke-width="1"
    >
      <g
        id="编组"
      >
        <g
          id="编组"
          transform="translate(36.000000, 0.000000)"
        >
          <path
            d="M63,9 L63,24 L78,24 L78,84 C78,85.6568542 76.6568542,87 75,87 L21,87 C19.3431458,87 18,85.6568542 18,84 L18,12 C18,10.3431458 19.3431458,9 21,9 L63,9 Z"
            fill="#E2E8F3"
            id="路径"
          />
          <rect
            fill="#CAD0DB"
            height="9"
            id="矩形"
            width="9"
            x="24"
            y="24"
          />
          <polygon
            fill="#EEF4FF"
            id="矩形"
            points="63 9 78 24 63 24"
          />
          <rect
            fill="#CAD0DB"
            height="6"
            id="矩形"
            width="48"
            x="24"
            y="39"
          />
          <rect
            fill="#CAD0DB"
            height="6"
            id="矩形"
            width="24"
            x="24"
            y="51"
          />
        </g>
        <g
          fill="url(#kd-empty-default-linearGradient3)"
          id="背景/云"
          opacity="0.300000012"
          transform="translate(108.000000, 40.000000)"
        >
          <g
            id="形状结合"
          >
            <path
              d="M3,39 L3,24 C3,15.7157288 9.71572875,9 18,9 C25.4616776,9 31.6508469,14.4482558 32.8063948,21.5836543 C33.797875,21.2064779 34.8748247,21 36,21 C40.9705627,21 45,25.0294373 45,30 L45,39 L3,39 Z"
            />
          </g>
        </g>
        <g
          fill="url(#kd-empty-default-linearGradient3)"
          id="背景/云"
          opacity="0.300000012"
          transform="translate(12.000000, 32.000000)"
        >
          <g
            id="形状结合"
            transform="translate(24.000000, 24.000000) scale(-1, 1) translate(-24.000000, -24.000000) "
          >
            <path
              d="M3,39 L3,24 C3,15.7157288 9.71572875,9 18,9 C25.4616776,9 31.6508469,14.4482558 32.8063948,21.5836543 C33.797875,21.2064779 34.8748247,21 36,21 C40.9705627,21 45,25.0294373 45,30 L45,39 L3,39 Z"
            />
          </g>
        </g>
      </g>
    </g>
  </svg>
  <div
    class="kd-empty-description"
  >
    暂无数据
  </div>
</div>
`;

exports[`Empty renders correctly 2`] = `
<div
  class="kd-empty"
>
  <svg
    class="kd-empty-image"
    height="96px"
    version="1.1"
    viewBox="0 0 168 96"
    width="168px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      编组
    </title>
    <defs>
      <linearGradient
        id="kd-empty-default-linearGradient4"
        x1="50%"
        x2="50%"
        y1="0%"
        y2="100%"
      >
        <stop
          offset="0%"
          stop-color="#D6DCE7"
        />
        <stop
          offset="100%"
          stop-color="#D6DCE7"
          stop-opacity="0"
        />
      </linearGradient>
    </defs>
    <g
      fill="none"
      fill-rule="evenodd"
      id="页面-1"
      stroke="none"
      stroke-width="1"
    >
      <g
        id="编组"
      >
        <g
          id="编组"
          transform="translate(36.000000, 0.000000)"
        >
          <path
            d="M63,9 L63,24 L78,24 L78,84 C78,85.6568542 76.6568542,87 75,87 L21,87 C19.3431458,87 18,85.6568542 18,84 L18,12 C18,10.3431458 19.3431458,9 21,9 L63,9 Z"
            fill="#E2E8F3"
            id="路径"
          />
          <rect
            fill="#CAD0DB"
            height="9"
            id="矩形"
            width="9"
            x="24"
            y="24"
          />
          <polygon
            fill="#EEF4FF"
            id="矩形"
            points="63 9 78 24 63 24"
          />
          <rect
            fill="#CAD0DB"
            height="6"
            id="矩形"
            width="48"
            x="24"
            y="39"
          />
          <rect
            fill="#CAD0DB"
            height="6"
            id="矩形"
            width="24"
            x="24"
            y="51"
          />
        </g>
        <g
          fill="url(#kd-empty-default-linearGradient4)"
          id="背景/云"
          opacity="0.300000012"
          transform="translate(108.000000, 40.000000)"
        >
          <g
            id="形状结合"
          >
            <path
              d="M3,39 L3,24 C3,15.7157288 9.71572875,9 18,9 C25.4616776,9 31.6508469,14.4482558 32.8063948,21.5836543 C33.797875,21.2064779 34.8748247,21 36,21 C40.9705627,21 45,25.0294373 45,30 L45,39 L3,39 Z"
            />
          </g>
        </g>
        <g
          fill="url(#kd-empty-default-linearGradient4)"
          id="背景/云"
          opacity="0.300000012"
          transform="translate(12.000000, 32.000000)"
        >
          <g
            id="形状结合"
            transform="translate(24.000000, 24.000000) scale(-1, 1) translate(-24.000000, -24.000000) "
          >
            <path
              d="M3,39 L3,24 C3,15.7157288 9.71572875,9 18,9 C25.4616776,9 31.6508469,14.4482558 32.8063948,21.5836543 C33.797875,21.2064779 34.8748247,21 36,21 C40.9705627,21 45,25.0294373 45,30 L45,39 L3,39 Z"
            />
          </g>
        </g>
      </g>
    </g>
  </svg>
  <div
    class="kd-empty-description"
  >
    我是空组件
  </div>
</div>
`;

exports[`Empty renders correctly 3`] = `
<div
  class="kd-empty"
>
  <img
    alt="empty"
    class="kd-empty-image"
    src="http://ikd.kingdee.com/ikd2/images/common/empty-content.png"
  />
  <div
    class="kd-empty-description"
  >
    暂无数据
  </div>
</div>
`;

exports[`Empty renders correctly 4`] = `
<div
  class="kd-empty"
>
  <svg
    class="kd-empty-image"
    height="96px"
    version="1.1"
    viewBox="0 0 168 96"
    width="168px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      编组
    </title>
    <defs>
      <linearGradient
        id="kd-empty-default-linearGradient5"
        x1="50%"
        x2="50%"
        y1="0%"
        y2="100%"
      >
        <stop
          offset="0%"
          stop-color="#D6DCE7"
        />
        <stop
          offset="100%"
          stop-color="#D6DCE7"
          stop-opacity="0"
        />
      </linearGradient>
    </defs>
    <g
      fill="none"
      fill-rule="evenodd"
      id="页面-1"
      stroke="none"
      stroke-width="1"
    >
      <g
        id="编组"
      >
        <g
          id="编组"
          transform="translate(36.000000, 0.000000)"
        >
          <path
            d="M63,9 L63,24 L78,24 L78,84 C78,85.6568542 76.6568542,87 75,87 L21,87 C19.3431458,87 18,85.6568542 18,84 L18,12 C18,10.3431458 19.3431458,9 21,9 L63,9 Z"
            fill="#E2E8F3"
            id="路径"
          />
          <rect
            fill="#CAD0DB"
            height="9"
            id="矩形"
            width="9"
            x="24"
            y="24"
          />
          <polygon
            fill="#EEF4FF"
            id="矩形"
            points="63 9 78 24 63 24"
          />
          <rect
            fill="#CAD0DB"
            height="6"
            id="矩形"
            width="48"
            x="24"
            y="39"
          />
          <rect
            fill="#CAD0DB"
            height="6"
            id="矩形"
            width="24"
            x="24"
            y="51"
          />
        </g>
        <g
          fill="url(#kd-empty-default-linearGradient5)"
          id="背景/云"
          opacity="0.300000012"
          transform="translate(108.000000, 40.000000)"
        >
          <g
            id="形状结合"
          >
            <path
              d="M3,39 L3,24 C3,15.7157288 9.71572875,9 18,9 C25.4616776,9 31.6508469,14.4482558 32.8063948,21.5836543 C33.797875,21.2064779 34.8748247,21 36,21 C40.9705627,21 45,25.0294373 45,30 L45,39 L3,39 Z"
            />
          </g>
        </g>
        <g
          fill="url(#kd-empty-default-linearGradient5)"
          id="背景/云"
          opacity="0.300000012"
          transform="translate(12.000000, 32.000000)"
        >
          <g
            id="形状结合"
            transform="translate(24.000000, 24.000000) scale(-1, 1) translate(-24.000000, -24.000000) "
          >
            <path
              d="M3,39 L3,24 C3,15.7157288 9.71572875,9 18,9 C25.4616776,9 31.6508469,14.4482558 32.8063948,21.5836543 C33.797875,21.2064779 34.8748247,21 36,21 C40.9705627,21 45,25.0294373 45,30 L45,39 L3,39 Z"
            />
          </g>
        </g>
      </g>
    </g>
  </svg>
  <div
    class="kd-empty-description"
  >
    暂无数据
  </div>
  <div
    class="kd-empty-footer"
  >
    空组件自定义附属内容
  </div>
</div>
`;
